app/src/main/java/com/github/steroidteam/todolist/model/todo/Task.java

```diff
@@ -65,4 +70,17 @@ public Boolean isDone() {
     public void setDone(Boolean done) {
         this.done = done;
     }
+
+    public void setDueDate(Date dueDate) {
+        Objects.requireNonNull(dueDate);
+        this.dueDate = dueDate;
+    }
+
+    public void removeDueDate() {
+        this.dueDate = null;
+    }
+
+    public Date getDueDate() {
```

Comment:
I would suggest to go "functionnal". Why not return an Optional<Date> since the dueDate can be null ? This forces us to deal with both cases and makes our code less error-prone...

Reply:
It's just a matter of consistency; as you can see the `done` attribute could be `null` as well but we do not do such a thing.

I agree that it could be useful to do such a thing, but in that case I believe we should apply the same pattern to all of our models in a task of their own (because otherwise we are going to have inconsistent APIs).

What do you think?

Reply:
You're right. In the current state of the project, maybe this issue is out of the scope of this PR. If the rest of the team agrees, we could create a new refactoring task in an upcoming sprint to rework the way we handle these details. I just want to make sure we don't create technical dept and make our development harder in the future whenever some null value is not handled correctly.